VimeoNetworking/VimeoNetworking/NSError+Extensions.swift

```diff
@@ -86,22 +101,52 @@ public extension NSError
         
         return error
     }
-
+    
+    /**
+     Adds a domain to a preexisting error
+     
+     - parameter domain: New domain for the error
+     
+     - returns: The error with additional information in the user info dictionary
+     */
     func errorByAddingDomain(domain: String) -> NSError
     {
         return self.errorByAddingDomain(domain, code: nil, userInfo: nil)
     }
     
+    /**
+     Adds user info to a preexisting error
+     
+     - parameter userInfo: the user info dictionary to append to the existing user info
+    
+     - returns: The error with additional user info
+     */
     func errorByAddingUserInfo(userInfo: [String: AnyObject]) -> NSError
     {
         return self.errorByAddingDomain(nil, code: nil, userInfo: userInfo)
     }
     
+    /**
+     Adds an error code to a preexisting error
+     
+     - parameter code: the new error code for the error
+     
+     - returns: The error with additional information in the user info dictionary
+     */
     func errorByAddingCode(code: Int) -> NSError
     {
         return self.errorByAddingDomain(nil, code: code, userInfo: nil)
     }
     
+    /**
+     Adds information to a preexisting error
+     
+     - parameter domain:   New domain for the error
+     - parameter code:     the new error code for the error
+     - parameter userInfo: the user info dictionary to append to the existing user info
+     
+     - returns: The error with additional information in the user info dictionary
```

Comment:
Worth mentioning that this and other methods in this class return a new error object rather than augmenting the existing one?